catroid/src/androidTest/java/org/catrobat/catroid/test/io/RemixTest.java

```diff
@@ -0,0 +1,139 @@
+		assertEquals("Unexpected remix-url header-field!",
+				expectedUrlFieldValue, mergedProject.getXmlHeader().getRemixParentsUrlString());
+	}
+
+	@Test
+	public void testVerifyRemixUrlsOfMergedProgramWhereFirstProgramHasBeenDownloadedAndSecondProgramHasBeenLocallyCreated() throws Exception {
+		Project mergedProject = new Project();
+		createProjectWithDifferentScripts();
+		String expectedUrlOfFirstProgram = "http://pocketcode.org/details/3218";
+		firstProject.getXmlHeader().setRemixParentsUrlString(expectedUrlOfFirstProgram);
+
+		ProjectManager.getInstance().createHeader(mergedProject, firstProject, secondProject);
+
+		String expectedUrlFieldValue = String.format("%s [%s], %s",
+				firstProject.getName(), expectedUrlOfFirstProgram, secondProject.getName());
+
+		String mergedRemixOfString = (String) Reflection.getPrivateField(XmlHeader.class,
+				mergedProject.getXmlHeader(), "remixGrandparentsUrlString");
+
+		assertEquals("Expecting remixOf header-field to be empty!", "", mergedRemixOfString);
+		assertEquals("Unexpected remix-url header-field!",
+				expectedUrlFieldValue, mergedProject.getXmlHeader().getRemixParentsUrlString());
+	}
+
+	@Test
+	public void testVerifyRemixUrlsOfMergedProgramWhereFirstProgramHasBeenLocallyCreatedAndSecondProgramHasBeenDownloaded() throws Exception {
+		Project mergedProject = new Project();
+
+		createProjectWithDifferentScripts();
+		String expectedUrlOfSecondProgram = "http://pocketcode.org/details/3218";
+		secondProject.getXmlHeader().setRemixParentsUrlString(expectedUrlOfSecondProgram);
+
+		ProjectManager.getInstance().createHeader(mergedProject, firstProject, secondProject);
+
+		String expectedUrlFieldValue = String.format("%s, %s [%s]", firstProject.getName(),
+				secondProject.getName(), expectedUrlOfSecondProgram);
+
+		String mergedRemixOfString = (String) Reflection.getPrivateField(XmlHeader.class,
+				mergedProject.getXmlHeader(), "remixGrandparentsUrlString");
+
+		assertEquals("Expecting remixOf header-field to be empty!", "", mergedRemixOfString);
+		assertEquals("Unexpected remix-url header-field!",
+				expectedUrlFieldValue, mergedProject.getXmlHeader().getRemixParentsUrlString());
+	}
+
+	private void createProjectWithDifferentScripts() {
+		XstreamSerializer storageHandler = XstreamSerializer.getInstance();
+
+		String firstSpriteName = getContext().getResources().getString(R.string.default_project_cloud_sprite_name_1);
+		String secondSpriteName = getContext().getResources().getString(R.string.default_project_cloud_sprite_name_2);
+
+		firstProject = TestUtils.createProjectWithGlobalValues("First Project", firstSpriteName, "test1", getContext());
+		secondProject = TestUtils.createProjectWithSpriteValues("Second Project", firstSpriteName, "test2", getContext());
+		secondProject.getDefaultScene().addSprite(new SingleSprite(secondSpriteName));
+		storageHandler.getInstance().saveProject(firstProject);
+		storageHandler.getInstance().saveProject(secondProject);
+	}
+}
```

Comment:
in general, please move the utils methods over here, and try to do a test data setup in a clear and easy to read manner.
do not mix multiple ways of accessing the same object, and try to keep everything clear and linear for the reader.

Reply:
if this is supposed to check the remix graph merge, there should be no variables and sprites required for the test.